app/templating/view_context.py

```diff
@@ -201,25 +200,32 @@ def _remove_unwanted_questions_answers(block, answers_in_block, answer_ids_to_ke
     return reduced_block
 
 
-def _get_formatted_total(block, answer_store, group_instance, schema):
+def _get_formatted_total(groups):
     calculated_total = 0
-    for answer in block['calculation']['answers_to_calculate']:
-        if group_instance > 0 and not schema.answer_is_in_repeating_group(answer):
-            group_instance = 0
-        calculated_total += get_answer_store_value(answer, answer_store, group_instance) or 0
-
-    answer_json = schema.get_answer(block['calculation']['answers_to_calculate'][0])
-    number_type = answer_json['type']
-    if number_type == 'Currency':
-        formatted_total = format_currency(calculated_total, answer_json['currency'])
-    elif number_type == 'Unit':
-        formatted_total = format_unit(answer_json['unit'], calculated_total)
-    elif number_type == 'Percentage':
-        formatted_total = format_percentage(calculated_total)
-    else:
-        formatted_total = format_number(calculated_total)
-
-    return formatted_total
+    answer_format = None
+    for group in groups:
+        for block in group['blocks']:
+            for question in block['questions']:
+                for answer in question['answers']:
+                    if not answer_format:
+                        answer_format = {
+                            'type': answer['type'],
+                            'unit': answer.get('unit'),
+                            'currency': answer.get('currency'),
+                        }
+                    answer_value = answer.get('value') or 0
```

Comment:
Perhaps answer.get('value', 0 ) is easier to read ?

Reply:
That's what I originally had but doesn't work if answer['value'] == None. This way of doing it not only ensure the key is there but also that a value is assigned against that key.